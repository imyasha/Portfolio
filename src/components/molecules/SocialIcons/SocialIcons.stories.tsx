import { SocialIcons } from './SocialIcons';

export default {
	title: 'molecules/SocialIcons',
	component: SocialIcons,
};

export const SocialIconsStory = () => <SocialIcons />;

SocialIconsStory.storyName = 'SocialIcons';
