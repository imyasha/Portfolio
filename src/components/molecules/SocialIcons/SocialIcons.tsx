import { Icon, IconProps } from 'Atoms/Icon';

export interface SocialIconsProps {
	profiles: Array<{
		url: string;
		name: string;
		icon: IconProps['icon'];
	}>;
}

const SocialIcons = ({ profiles }: SocialIconsProps) => {
	return (
		<ul className="flex gap-6">
			{profiles.map(({ url, name, icon }) => (
				<li
					className="w-6 h-6 opacity-70 hover:opacity-100 transition"
					key={name}
				>
					<a href={url} target="_blank" rel="noopener noreferrer" title={name}>
						<Icon icon={icon} />
					</a>
				</li>
			))}
		</ul>
	);
};

export { SocialIcons };
