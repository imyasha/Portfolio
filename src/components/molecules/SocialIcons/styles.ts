import styled from 'styled-components';

const StyledSocialIcons = styled.ul`
	display: flex;

	li {
		width: 1.5rem;
		margin: 0 0.5rem;
	}

	a {
		color: ${({ theme }) => theme.text};
		text-decoration: none;
		transition: 0.2s;

		&:hover {
			color: ${({ theme }) => theme.headline};
		}
	}

	@media screen and (min-width: 768px) {
		li {
			margin-left: 1rem;
		}
	}
`;

export { StyledSocialIcons };
