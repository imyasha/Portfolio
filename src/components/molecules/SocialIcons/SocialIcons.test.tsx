import { render } from '@testing-library/react';
import { SocialIcons } from '../SocialIcons';

describe('SocialIcons', () => {
	it('renders correctly', () => {
		const { container } = render(<SocialIcons />);
		expect(container).toMatchSnapshot();
	});
});
