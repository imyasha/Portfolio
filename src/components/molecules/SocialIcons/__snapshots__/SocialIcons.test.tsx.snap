// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialIcons renders correctly 1`] = `
<div>
  <ul
    class="styles__StyledSocialIcons-sc-15vq785-0 iHrrTj"
  >
    <li>
      <a
        href="https://github.com/jakeherp"
        rel="noopener noreferrer"
        target="_blank"
        title="Github"
      >
        <test-file-stub />
      </a>
    </li>
    <li>
      <a
        href="https://twitter.com/jakeherp"
        rel="noopener noreferrer"
        target="_blank"
        title="Twitter"
      >
        <test-file-stub />
      </a>
    </li>
    <li>
      <a
        href="https://www.linkedin.com/in/jacobherper"
        rel="noopener noreferrer"
        target="_blank"
        title="LinkedIn"
      >
        <test-file-stub />
      </a>
    </li>
    <li>
      <a
        href="https://instagram.com/jakeherp"
        rel="noopener noreferrer"
        target="_blank"
        title="Instagram"
      >
        <test-file-stub />
      </a>
    </li>
  </ul>
</div>
`;
